### DISCUSSION TITLE
Help with my fork, I want to revert

### DISCUSSION THREAD

**Discussion by mikygee** [community/community#28652] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello,

I forked another repository and update code from time to time.

However today I messed up, the GUI shows me I'm 6 commits ahead but I want to cancel everything on that GUI
![image](https://user-images.githubusercontent.com/2222550/183288376-46fa37bb-79f6-41de-a46a-4216cb1aef9c.png)

These 6 commits are do and undoes. I'm still in sync with the other person I forked.
The commits are like add "this" in file X, then the other commit is remove "this" in file X
I don't want to merge my mess with the other person.

All I want is to go back to where I was this morning and remove "6 commits ahead"

How can I do that I'm lost.

Thank you

**Comment by melvincwng** [comment]
@mikygee , perhaps you can try deleting that branch/fork if the changes/commits are not needed?

Then refork the original repository if necessary to get a fresh start.

Hope this helps.

**Comment by mikygee** [comment]
Thank you for your quick answer.
Do you know if I will loose all my statistics ?

**Comment by melvincwng** [comment]
@mikygee, if you are referring to the contributions in the 'Contributions graph' of your GitHub Profile page - then yes, it should be removed.

**Reply by melvincwng** [reply]
@mikygee , alternatively why not create a new branch from the master repo?

Then make your changes in the new branch...then merge into the original repository. 

That way you get to keep your existing commits in the contribution graph, and have a new clean slate to work with.

**Comment by airtower-luna** [comment]
If you don't want to delete the fork, you can work with the command line: `git reset` your branch to the same commit as the upstream branch, and force push. The "force" is necessary because you'll lose the commits you have added, and by default Git avoids that.

**Comment by mikygee** [comment, marked as answer]
Hello everyone,
Thank you for all your answers.
To solve the problem I did
```
# git reset --hard 5dc20c9
# git push --force https://xxxx@github.com/mikygee/immobilier.git
```
5dc20c9 is the point where I wanted to restore.
Thanks to you I solved in less than 10 minutes what I have been searching for 2 hours.

**Reply by ThienVie** [reply]
Hello, what does these xxxx from "https://xxxx@github.com/..." mean? I am stuck at this.

**Reply by mikygee** [reply]
It's my token, it's secret that's why I replaced it with xxxx

**Reply by airtower-luna** [reply]
Adding the token to the URL is highly discouraged, it creates a security risk: Your token gets stored on disk in plain text, and it tends to show up in output. If you don't want to copy & paste (or worse, type) it every time look at [Git Tools – Credential Storage](https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage). My personal preference is the `cache` method described there, plus a password manager ([KeePassXC](https://keepassxc.org/)) for long-term storage of the token. :slightly_smiling_face:

**Reply by mikygee** [reply]
Please tell me where is the security risk:
- My command history is lost when I logout
- I'm the only user of my machine
- Token in either way is sent by https

But yes I agree if you're on a shared machine, a vm and your shell history is stored then someone could find it out.
Thank you for this best practice.

**Reply by airtower-luna** [reply]
> Please tell me where is the security risk

* Stolen disk. Obviously this risk is much higher for mobile devices (like a laptop), and you can mostly mitigate it by using full disk encryption.
* Security issues along the line of this: https://blog.mozilla.org/security/2015/08/06/firefox-exploit-found-in-the-wild/ In that case the attacker could read arbitrary files from a malicious PDF, but not execute random code.
* With random code execution all bets are off, but encrypted storage (like in a decent password manager) at least makes it harder for the attacker: They can only break it if they gain enough privileges to use a keylogger or read the data from memory while unencrypted.
